packages/components/toggle/src/Toggle.tsx

```diff
@@ -0,0 +1,56 @@
+import type { ToggleProps } from './toggle.types';
+import {
+  ToggleStyled,
+  ToggleContainerStyled,
+  SpanStyled,
+  InputStyled,
+} from './toggle.styles';
+import { useTheme } from '@douro-ui/react';
+import React, { ChangeEvent } from 'react';
+
+const Toggle = ({
+  disabled,
+  checked,
+  onChange,
+  onToggleChange,
+  ...props
+}: ToggleProps & {
+  onToggleChange?: (checked: boolean) => void;
+}): React.ReactNode => {
+  const theme = useTheme();
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (!disabled) {
+      onChange?.(event);
+      onToggleChange?.(event.target.checked);
+    }
+  };
+
+  const handleClick = () => {
+    if (!disabled) {
+      onToggleChange?.(!checked);
+    }
+  };
+
+  return (
+    <ToggleContainerStyled className={disabled ? 'disabled' : ''}>
```

Comment:
Playing with conditions with Emotion is much easier and you don't need to use classnames.
Pass a prop named disable, and use it conditional in styles to return the CSS you when its active or not.

```suggestion
    <ToggleContainerStyled disabled>
```